proxy/logging/LogField.cc

```diff
@@ -772,7 +772,9 @@ LogFieldList::marshal_len(LogAccess *lad)
   int bytes = 0;
   for (LogField *f = first(); f; f = next(f)) {
     if (f->type() != LogField::sINT) {
-      bytes += f->marshal_len(lad);
+      const int len = f->marshal_len(lad);
```

Comment:
The core of the problem was LogAccessHttp.cc:ln487,
`      len = round_strlen(m_client_req_unmapped_url_host_len + 1); // +1 for eos`
When m_client_req_unmapped_url_host_len is -1, len is set to 0.
And strlen of -1 is kind of gross way to indicate that can't resolve the host name. In the future we would like to change all these to StringView, so this nudges us in that direction. 

This assert is just to ensure no other code paths where having the same issue.